### PR TITLE
Add ReportTemplate class

### PR DESCRIPTION
**Pull request by lhellebr**

### CODE REVIEW

**Comment by ntkathole** [comment]
[test]

**Comment by mirzal** [comment]
@lhellebr do you think we should merge this before creating 6.5.z branch?

Seeing how it depends on PRs from Robottelo and Airgun that didn't make it into 6.5.z branches there, I guess answer is "no", but I would like confirmation.

**Comment by lhellebr** [comment]
@mirzal This only contains 6.5 functionality so this should be both in master and in 6.5 branch. This PR doesn't depend on anything, rather other PRs depend on this one. I don't have a preference about when to merge this as long as it doesn't cause blocking dependent PRs.

**Comment by mirzal** [comment]
@lhellebr thanks, makes sense.

Why is it in Draft state? Is there anything else that you would like to add before making it non-Draft?
We can't ACK and merge Draft PRs.

**Comment by lhellebr** [comment]
@mirzal I will flip it to non-draft once I finish work on https://github.com/SatelliteQE/robottelo/pull/6878 (I am not sure if I won't need to add something).
Or, if you prefer, I can flip it to non-draft now and open a new PR if I need to add something.

**Comment by mirzal** [comment]
@lhellebr OK, so we shouldn't really merge this one until https://github.com/SatelliteQE/robottelo/pull/6878 is in shape (non-Draft). As long as it's not merged, you can push new commits here, no need to open new PRs. Let's leave it in Draft mode for now and toggle both this and Robottelo to non-Draft at the same time.

**Comment by mirzal** [comment]
@lhellebr we branched 6.5.z today, so this PR didn't make it in. Once it's merged into master, feel free to send cherry-pick PR for 6.5.z.

@san7ket @oshtaier @rplevka please review it on your earliest convenience.

File: nailgun/entities.py [hunk 1]
```diff
@@ -1810,6 +1810,93 @@ def clone(self, synchronous=True, **kwargs):
...
+            'name': entity_fields.StringField(
+                required=True,
+                str_type='alpha',
+                length=(6, 12),
+                unique=True
+            ),
+            'organization': entity_fields.OneToManyField(Organization),
+            'location': entity_fields.OneToManyField(Location),
+            'template': entity_fields.StringField(required=True),
+            'default': entity_fields.StringField(required=True),
```
Comment (oshtaier) [inline nailgun/entities.py]: `Must be one of: true, false, 1, 0.` sounds as Boolean type

File: nailgun/entities.py [hunk 2]
```diff
@@ -1810,6 +1810,93 @@ def clone(self, synchronous=True, **kwargs):
...
+
+        clone
+            /report_templates/clone
+        revision
+            /report_templates/revision
+
+        ``super`` is called otherwise.
+
+        """
+        if which in ('clone', 'revision'):
```
Comment (oshtaier) [inline nailgun/entities.py]: whether remove `revision` or implement helper for it

**Comment by oshtaier** [comment]
Pending:
1) my first comment, second is just a nitpick
2) Test result for create, update, delete or robottelo PR with such results
3) Unit tests are mandatory for nailgun framework

**Comment by lhellebr** [comment]
I am working on unit tests, otherwise the comments were addressed

**Comment by lhellebr** [comment]
```
$ pytest test_reporttemplates.py
========================================= test session starts ==========================================
platform linux -- Python 3.6.8, pytest-3.6.1, py-1.7.0, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/lhellebr/git/robottelo, inifile:
plugins: services-1.3.0, mock-1.10.0
collecting 5 items                                                                                     2019-05-23 15:39:58 - conftest - DEBUG - Collected 5 test cases

collected 5 items                                                                                      

test_reporttemplates.py sss..                                                                    [100%]

================================ 2 passed, 3 skipped in 527.09 seconds =================================
```

**Comment by lhellebr** [comment]
I am not able to run unittests without errors (without any of my changes). When running ``make test``  with Python2, I get a lot of unicode vs str errors, like ``AttributeError: 'str' object has no attribute 'isnumeric'`` while with Python3 (changing python to python3 in Makefile), I keep getting ``ImportError: cannot import name 'BaseDirectory'`` (I DO have xdg module installed). Any advices on how to run unit tests properly?

**Comment by lhellebr** [comment]
Tests added

**Comment by lhellebr** [comment]
@mirzal @oshtaier @rplevka @san7ket These PRs have been open for too long, can we finally merge this?

**Comment by oshtaier** [comment]
1 and 3 points are addressed, but can't see test results for api calls. Not sure what https://github.com/SatelliteQE/nailgun/pull/629#issuecomment-495226727 is. Do these results refer to API test cases? Do you have calls for `create`, `update` and `delete` there?

**Comment by lhellebr** [comment]
You're right, some calls were missing. Here is the results of running a test from PR https://github.com/SatelliteQE/robottelo/pull/6973

```
$ pytest tests/foreman/api/test_reporttemplates.py::ComputeResourceTestCase::test_positive_e2e_crud
=============================================== test session starts ===============================================
platform linux -- Python 3.6.8, pytest-3.6.1, py-1.7.0, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/lhellebr/git/robottelo, inifile:
plugins: services-1.3.0, mock-1.10.0
collecting 1 item                                                                                                 2019-05-30 15:18:55 - conftest - DEBUG - Collected 1 test cases

collected 1 item                                                                                                  

tests/foreman/api/test_reporttemplates.py .                                                                 [100%]

============================================ 1 passed in 6.52 seconds =============================================
```

**Comment by oshtaier** [comment]
@mirzal or anyone else let's merge current PR

**Comment by mirzal** [comment]
Since Robottelo master now uses Nailgun `stable-satellite` branch, I think we need PR with cherry-pick.

**Pull request merged**